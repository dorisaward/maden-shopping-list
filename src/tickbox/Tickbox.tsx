import React from 'react';
import {StyleSheet, View} from 'react-native';

type Props = {
  ticked: boolean;
};

export const Tickbox = ({ticked}: Props): React.JSX.Element => (
  <View style={tickStyles.box} testID='tickbox'>
    {ticked && (
      <>
        <View style={tickStyles.stem} />
        <View style={tickStyles.kick} />
      </>
    )}
  </View>
);

const tickStyles = StyleSheet.create({
  box: {
    width: 20,
    height: 20,
    borderColor: 'black',
    borderWidth: 1,
    marginVertical: 10,
    marginHorizontal: 15,
  },
  stem: {
    transform: 'rotate(45deg)',
    position: 'absolute',
    width: 1,
    height: 16,
    backgroundColor: 'black',
    left: 12,
    top: 0,
  },
  kick: {
    transform: 'rotate(315deg)',
    position: 'absolute',
    width: 1,
    height: 8,
    left: 4,
    top: 8,
    backgroundColor: 'black',
  },
});
