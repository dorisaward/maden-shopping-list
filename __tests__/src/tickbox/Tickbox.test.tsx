import 'react-native';
import React from 'react';
import {it, expect} from '@jest/globals';
import {render} from '@testing-library/react-native';
import {Tickbox} from '../../../src/tickbox/Tickbox';

it.each<boolean>([true, false])('should render, given %p', ticked => {
  // Given
  const renderable = <Tickbox ticked={ticked} />;

  // When
  const {toJSON} = render(renderable);

  // Then
  expect(toJSON()).toMatchSnapshot();
});
