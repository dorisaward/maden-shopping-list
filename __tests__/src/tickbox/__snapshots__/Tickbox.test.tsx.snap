// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render, given false 1`] = `
<View
  style={
    {
      "borderColor": "black",
      "borderWidth": 1,
      "height": 20,
      "marginHorizontal": 15,
      "marginVertical": 10,
      "width": 20,
    }
  }
  testID="tickbox"
/>
`;

exports[`should render, given true 1`] = `
<View
  style={
    {
      "borderColor": "black",
      "borderWidth": 1,
      "height": 20,
      "marginHorizontal": 15,
      "marginVertical": 10,
      "width": 20,
    }
  }
  testID="tickbox"
>
  <View
    style={
      {
        "backgroundColor": "black",
        "height": 16,
        "left": 12,
        "position": "absolute",
        "top": 0,
        "transform": "rotate(45deg)",
        "width": 1,
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "black",
        "height": 8,
        "left": 4,
        "position": "absolute",
        "top": 8,
        "transform": "rotate(315deg)",
        "width": 1,
      }
    }
  />
</View>
`;
